packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -11,13 +11,30 @@ import type {
   MultichainAccountWallet as MultichainAccountWalletDefinition,
 } from '@metamask/account-api';
 import type { AccountGroupId } from '@metamask/account-api';
-import type { AccountProvider } from '@metamask/account-api';
 import {
   type EntropySourceId,
   type KeyringAccount,
 } from '@metamask/keyring-api';
+import { createProjectLogger } from '@metamask/utils';
 
 import { MultichainAccountGroup } from './MultichainAccountGroup';
+import type { NamedAccountProvider } from './providers';
+
+/**
+ * The context for a provider discovery.
+ */
+export type AccountProviderDiscoveryContext = {
+  provider: NamedAccountProvider;
+  stopped: boolean;
+  groupIndex: number;
+  count: number;
+};
+
+export type AccountDiscoveryMetrics = {
+  [providerName: string]: number;
+};
+
+const log = createProjectLogger('multichain-account-service');
```

Comment:
Note for later: we will move this into a `logger.ts` file, so we can use it in other part of the service too.